mesonbuild/mesonmain.py

```diff
@@ -119,14 +120,22 @@ def run(self, args):
 
         # Hidden commands have their own parser instead of using the global one
         if args[0] in self.hidden_commands:
-            parser = self.commands[args[0]]
+            command = args[0]
+            parser = self.commands[command]
             args = args[1:]
         else:
             parser = self.parser
+            command = None
 
         args = mesonlib.expand_arguments(args)
         options = parser.parse_args(args)
 
+        if command is None:
+            command = options.command
+
+        if command in ('setup', 'compile', 'test', 'install') and sys.version_info < (3, 6):
```

Comment:
We can probably also add `'configure'` here.

Reply:
I thought about that, but then I thought that there might be some IDE plugin that's parsing `meson configure` output. The rest have more-or-less free-form output.

Reply:
the `meson configure` output is **not** supposed to be parsed (I already [told the eclipse devs](https://bugs.eclipse.org/bugs/show_bug.cgi?id=548350)). We have an introspection API for that.

Reply:
Yes, but the people using Python 3.5 are probably unable to update anything other than Meson itself. Arguably an unsupported configuration, but why take the chance. Anyone who runs `meson configure` is also going to run at least `setup` 🙂

Reply:
Just saying that we already had changes in the `meson configure` format and the only project that bothered constantly updating their fragile regex setup is/was eclipse...

Reply:
That's proof to me that we should not put the notice at the end of `configure`! 😄